backend/stakepoold/server.go

```diff
@@ -301,46 +485,304 @@ func main() {
+				continue
+			}
+		case "Stats":
+			// Always save stats
+			break
+		case "UserVotingConfig":
+			if len(ctx.userVotingConfig) == 0 {
+				log.Warn("saveData: UserVotingConfig is empty; skipping save")
+				continue
+			}
+		default:
+			log.Warn("saveData: passed unhandled data name " + filenameprefix)
+			continue
+		}
+
+		w, err := os.Create(destPath)
+		defer w.Close()
+		if err != nil {
+			log.Errorf("Error opening file %s: %v", ctx.dataPath, err)
+			continue
+		}
+
+		switch filenameprefix {
+		case "LiveTickets":
+			enc := gob.NewEncoder(w)
+			if err := enc.Encode(&ctx.liveTicketsMSA); err != nil {
+				log.Errorf("Failed to encode file %s: %v", ctx.dataPath, err)
+				continue
+			}
+		case "ManuallyAddedTickets":
+			enc := gob.NewEncoder(w)
+			if err := enc.Encode(&ctx.manuallyAddedTicketsMSA); err != nil {
+				log.Errorf("Failed to encode file %s: %v", ctx.dataPath, err)
+				continue
+			}
+		case "Stats":
+			enc := gob.NewEncoder(w)
+			if err := enc.Encode(&ctx.stats); err != nil {
+				log.Errorf("Failed to encode file %s: %v", ctx.dataPath, err)
+				continue
+			}
+		case "UserVotingConfig":
+			enc := gob.NewEncoder(w)
+			if err := enc.Encode(&ctx.userVotingConfig); err != nil {
+				log.Errorf("Failed to encode file %s: %v", ctx.dataPath, err)
+				continue
+			}
+		}
+
+		log.Infof("saveData: successfully saved %v data to %s",
+			filenameprefix, destPath)
+	}
+
 }
 
-// winner contains all the bits and pieces required to vote and to print
-// statistics after usage.
-type winner struct {
+// ticketMetadata contains all the bits and pieces required to vote new tickets,
+// to look up new/missed/spent tickets, and to print statistics after usage.
+type ticketMetadata struct {
+	blockHeight  int64
 	msa          string                    // multisig
 	ticket       *chainhash.Hash           // ticket
+	spent        bool                      // spent (true) or missed (false)
 	config       userdata.UserVotingConfig // voting config
-	signDuration time.Duration
-	sendDuration time.Duration
-	duration     time.Duration // overall vote duration
-	err          error         // log errors along the way
+	duration     time.Duration             // overall vote duration
+	getDuration  time.Duration             // time to gettransaction
+	txid         *chainhash.Hash           // transaction id
+	ticketType   string                    // new or spentmissed
+	signDuration time.Duration             // time to generatevote
+	sendDuration time.Duration             // time to sendrawtransaction
+	err          error                     // log errors along the way
+}
+
+// getticket pulls the transaction information for a ticket from dcrwallet. This is a go routine!
+func (ctx *appContext) getticket(wg *sync.WaitGroup, nt *ticketMetadata) {
+	waitStart := time.Now()
+
+	// getticket is used by processNewTickets and processSpentMissedTickets
+	// so we gracefully yield the processor until voting has completed
+	counter := 0
+	for {
```

Comment:
To avoid the atomic stuff with `ctx.lastVotedBlock` and this loop, there could be a gate keeper function that returns a future (or just a channel) that signals when the vote goroutine has handled the latest block and it's cool for `processNewTickets` and `processSpentMissedTickets` to do their thing.